File: Frontend/src/types/booking.ts
export type BookingStatus =
  | "PENDING_PAYMENT"
  | "CONFIRMED"
  | "COMPLETED"
  | "CANCELLED_BY_USER"
  | "CANCELLED_BY_OWNER"
  | "EXPIRED"
  | "NO_SHOW";

export type BookingPaymentStatus = "UNPAID" | "PAID" | "FAILED" | "REFUNDED";
export type BookingType = "ONLINE" | "OFFLINE" | "ADMIN";

export type BookingSource =
  | "MATCHOP"
  | "ZALO"
  | "FACEBOOK"
  | "PHONE"
  | "DIRECT"
  | "OTHER";

export interface Booking {
  id: string;
  userId?: string;
  ownerId?: string;
  venueId: string;
  courtId: string;
  sportId: string;
  bookingDate: string;
  startTime: string;
  endTime: string;
  totalPrice: number;
  status: BookingStatus;
  paymentStatus: BookingPaymentStatus;
  bookingType: BookingType;
  bookingSource?: BookingSource;
  customerName?: string;
  customerPhone?: string;
  expireAt?: string;
  note?: string;
  createdAt: string;
  venueName?: string;
  courtName?: string;
  sportName?: string;
}

export interface CourtCalendarEntry {
  bookingId?: string;
  blockId?: string;
  entryType: "BOOKING" | "BLOCK";
  startTime: string;
  endTime: string;
  status: string;
  bookingSource?: BookingSource;
  customerName?: string;
  note?: string;
}

export interface OwnerCourtCalendar {
  courtId: string;
  courtName: string;
  venueName: string;
  date: string;
  openingTime: string;
  closingTime: string;
  entries: CourtCalendarEntry[];
}

export interface CreateExternalBookingRequest {
  courtId: string;
  bookingDate: string;
  startTime: string;
  endTime: string;
  bookingSource: BookingSource;
  customerName: string;
  customerPhone: string;
  notes?: string;
}

export interface CreateBookingRequest {
  courtId: string;
  sportId: string;
  bookingDate: string;
  startTime: string;
  endTime: string;
  note?: string;
  customerName?: string;
  customerPhone?: string;
}

File: Backend/MATCHOP.API/Entities/Booking.cs
using MATCHOP.API.Enums;

namespace MATCHOP.API.Entities;

public class Booking
{
    public Guid Id { get; set; }
    public Guid? UserId { get; set; }
    public Guid? OwnerId { get; set; }
    public Guid VenueId { get; set; }
    public Guid CourtId { get; set; }
    public Guid SportId { get; set; }
    public DateOnly BookingDate { get; set; }
    public TimeOnly StartTime { get; set; }
    public TimeOnly EndTime { get; set; }
    public decimal TotalPrice { get; set; }
    public BookingStatus Status { get; set; } = BookingStatus.PENDING_PAYMENT;
    public BookingPaymentStatus PaymentStatus { get; set; } = BookingPaymentStatus.UNPAID;
    public BookingType BookingType { get; set; } = BookingType.ONLINE;
    public BookingSource BookingSource { get; set; } = BookingSource.MATCHOP;
    public string? CustomerName { get; set; }
    public string? CustomerPhone { get; set; }
    public DateTime? ExpireAt { get; set; }
    public string? Note { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties
    public User? User { get; set; }
    public User? Owner { get; set; }
    public Venue Venue { get; set; } = null!;
    public Court Court { get; set; } = null!;
    public Sport Sport { get; set; } = null!;
    public ICollection<BookingSlot> BookingSlots { get; set; } = new List<BookingSlot>();
    public ICollection<Payment> Payments { get; set; } = new List<Payment>();
}